StockAnalyzerService/StockAnalyzerService/Service/StockAnalyzer.cs

```diff
@@ -0,0 +1,29 @@
+﻿using StockAnalyzerService.Model;
+using StockAnalyzerService.Service;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace StockAnalyzerService.Service {
+	public class StockAnalyzer : IStockAnalyzer {
+		public IHttpClientFactory clientFactory;
+		public IHttpCalls httpCalls;
+		public StockAnalyzer(IHttpClientFactory _clientFactory, IHttpCalls _httpCalls) {
+			clientFactory = _clientFactory;
+			httpCalls = _httpCalls;
+		}
+
+		public async Task<List<StockMetadata>> GetStocksWithUsers() {
+			try {
+				string urlParameters = "stock/user";
+				HttpClient stockAnalyzerClient = clientFactory.CreateClient("stockAnalyzer");
+				return await httpCalls.Get<List<StockMetadata>>(stockAnalyzerClient, urlParameters);
+			} catch {
+				return new List<StockMetadata>();
```

Comment:
need to log exception if it occurrs?

Reply:
I logged an exception in the Get method, but I guess an exception could be thrown by CreateClient

Reply:
I think it throws a keynotfound exception if the httpclient with the name doesn't exist which is highly unlikely since we are hardcoding it in the configuration

Reply:
Done